src/blog/2024/06/calling-python-script-from-node-red.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Calling Python script from Node-RED
+subtitle: Guite on how to execute Python scripts from Node-RED
+description: Learn How to seamlessly execute python script from Node-RED.
+date: 2024-06-21
+authors: ["sumit-shinde"]
+image: 
+tags:
+   - post
+   - node-red
+   - how-to
+   - python
+---
+
+Have you ever needed to execute Python scripts from Node-RED? If so, this comprehensive guide is for you. We will walk you through the process of integrating Python seamlessly with Node-RED, alongside troubleshooting tips and techniques for passing arguments to scripts.
+
+<!--more-->
+
+## Why people use python with Node-RED
+
+Node-RED is a powerful visual programming tool used for building IoT applications. It is built on Node.js, which meets IoT application needs with its asynchronous operations and event-driven, non-blocking architecture. Javascript on which Node.js is built and the Node-RED supports can handle a wide range of tasks, However, Python excels in specific areas such as complex computations and machine learning due to its extensive set of libraries.
+
+Many users choose to integrate Python with Node-RED to leverage these strengths. By using Node-RED and Python together,  developers can create advanced IoT solutions that excel in computational tasks and analysis effectively.
+
+## Installing Python
+
+When executing Python scripts, it's essential to have the Python runtime installed on your system. However, if you've installed Node-RED using npm, Python is typically have already installed on your system. This is because Python is a prerequisite for npm to build native modules.
```

Comment:
```suggestion
When executing Python scripts, it's essential to have the Python runtime installed on your system. As Python is a prerequisite for installing Node-RED npm, you might have it already installed in your system. 
```

Reply:
No, it will not work in the ff cloud, https://github.com/FlowFuse/customer/issues/206#issuecomment-2083631535, but will work with the ff device agent.

Reply:
updated this suggestion for more clarity